### DISCUSSION TITLE
"Clearsigned file isn't valid" during apt update of Ubuntu 22.04 instance

### DISCUSSION THREAD

**Discussion by LucaGallinari** [community/community#120966] [Actions] [answered]
### Select Topic Area

Bug

### Body

Running "apt update" in an action with an Ubuntu 22.04 instance, leads to this error:

> E: Failed to fetch https://packages.microsoft.com/ubuntu/22.04/prod/dists/jammy/InRelease  Clearsigned file isn't valid, got 'NOSPLIT' (does the network require authentication?)
E: The repository 'https://packages.microsoft.com/ubuntu/22.04/prod jammy InRelease' is no longer signed.

![Screenshot 2024-04-24 alle 12 20 03](https://github.com/community/community/assets/5822554/a58655b6-436a-443e-aa57-7ace0638f1ed)

**Comment by RobinHoutevelts** [comment]
We are also seeing this

**Reply by johnrhunt** [reply]
Yeah, I'm hoping M$ fix this soon.

**Reply by drumst0ck** [reply]
Use  runs-on: ubuntu-20.04 instead of     runs-on: latest in your .yml

**Reply by xalelax** [reply]
@JoseSanchez54 This fixed it for me, thanks!

**Reply by LucaGallinari** [reply]
> Use runs-on: ubuntu-20.04 instead of runs-on: latest in your .yml

Should i mark this as an answer until properly addressed by MS?

**Reply by jazib194** [reply]
Hello! It seems like you are encountering an issue with a "Clearsigned file isn't valid" message during the apt update of your Ubuntu 22.04 instance. This error typically occurs when there is an issue with the signature verification of the repository's metadata.

To resolve this issue, you can try the following steps:
1. Make sure your system's date and time are correctly set.
2. Clear the apt cache by running the command: sudo apt clean.
3. Update the repository key by running: sudo apt-key adv --recv-keys --keyserver keyserver.ubuntu.com.
4. Try updating the package lists again by running: sudo apt update.

If the issue persists, you may need to check the repository configuration or contact the repository maintainer for further assistance. I hope this helps! Let me know if you need any more assistance for more information click [here](https://answertenant.com/real-estate/).

**Reply by ebndev** [reply]
Hi @JaLe29, thanks for wanting to participate in this Discussion! 

Please note: we’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.

**Comment by lyager** [comment]
Hit this too

**Comment by moffe42** [comment]
I think we all are....

**Reply by akshhack** [reply]
memes

**Reply by thuong41** [reply]
thinking

**Comment by fabien-ors** [comment]
Mee too :(

**Comment by ruslan-maistrenko** [comment]
too

**Comment by moffe42** [comment]
someone fåågd åb the latest Ubuntu release i guess....

**Comment by key-hui-mentem** [comment]
me too

**Comment by RobinHoutevelts** [comment]
🙈 Immediately regretted my comment.
Sorry for starting the mee too train 🚋

**Reply by Gewerd-Strauss** [reply]
It's the internet, and this was basically inevitable. It's somewhat amusing seeing people slowly arriving here one by one.

**Reply by lgmorand** [reply]
tchoo tchoo!! climbing on the metoo train

**Comment by jigneshgadhavi** [comment]
Yes I'm facing same issue with Github action all ubuntu images

**Comment by Bojan131** [comment]
I also have the same issue. It worked good yesterday.

**Reply by Claudius-Appel** [reply]
It worked well - or rather, it didn't break stuff - as recent as 3 hours ago. That was when my last action ran successfully P:

**Comment by aditya-mangosciences** [comment]
I checked this issue just when it was created, and thought its someone like me only to come 5 minutes laters seeing all these comments. 👍 
Thanks for confirming its affecting everyone.

**Comment by VilleFTW** [comment]
This must be addressed quickly. All of our builds are failing

**Comment by jypuigbo** [comment]
Same here...

**Comment by tarasvolosheniuk** [comment]
Same problem here. All builds are failing

**Comment by Niedman** [comment]
Same here...

**Comment by sthajas** [comment]
Same here :disappointed:

**Comment by kirugan** [comment]
Same ....

**Comment by 0Dmitry** [comment]
Same issue here

**Comment by vgoncearencu** [comment]
If you don't use any packages from the Microsoft repo, change from:
```
         sudo apt-get update
```

to:
```
         sudo rm /etc/apt/sources.list.d/microsoft-prod.list
         sudo apt-get update
```
in other case replace with `sudo apt-get update || true`

**Reply by lunatickochiya** [reply]
THX

**Reply by fabien-ors** [reply]
Thanks for the workaround which works for me :)

**Reply by jypuigbo** [reply]
my problem is that I need this file, since I'm installing **msodbcsql17** just after that

**Reply by johnrhunt** [reply]
This just removes all the available packages from Microsoft and might break your builds further.

My current line of thinking is to just wait for MS to fix their boo boo.

**Reply by vgoncearencu** [reply]
For the devs that uses some packages from the Microsft repo, change from `sudo apt-get update` to `sudo apt-get update || true`

**Reply by jypuigbo** [reply]
This is not a solution. After that I can't `apt-get install -y msodbcsql17`, and therefore rebuild my docker...

**Reply by LucaGallinari** [reply]
I marked this as a "workaround" answer until MS address this

**Reply by npenateca** [reply]
There is any health page to monitor when MSFT fix the issue?. This workarround doesn't work for me as i need to install msodbsql18

**Reply by deavmi** [reply]
Many thanks sir :+1:

**Reply by dobladov** [reply]
@npenateca you can check here when it's fixed https://www.githubstatus.com/

**Reply by LucaGallinari** [reply]
@npenateca There is another workaround here https://github.com/orgs/community/discussions/120966?sort=top#discussioncomment-9212201

**Comment by imraax** [comment]
Same issue here just now

**Comment by anuragpindoriya** [comment]
Is this permanent issue or temporary issue  ???

**Comment by Niedman** [comment]
I've removed:
```
sudo apt-get update -y
sudo apt-get install -y google-cloud-sdk-gke-gcloud-auth-plugin
```

Replaced it for: 
`gcloud components install gke-gcloud-auth-plugin`

and it works! I'm still not sure if it is the correct way though.

I've follow this article: https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke

**Comment by wojciech-jakubowski** [comment]
#metoo

**Comment by SokoloffA** [comment]
This URL returns an empty file
https://packages.microsoft.com/ubuntu/22.04/prod/dists/jammy/InRelease

**Comment by damianopetrungaro** [comment]
Same here.

**Comment by Ganeshraut2141** [comment]
same here looking workaround

**Comment by Kirill89** [comment]
Same here. We temporary removed `sudo apt-get update` as a workaround.

**Reply by Niedman** [reply]
I've tried it and it didn't work for me.

**Reply by Kirill89** [reply]
@Niedman what was the error without `sudo apt-get update`?

**Reply by Niedman** [reply]
@Kirill89 I don't think my issue is quite exactly this one but it's a different flavour. 
```
Reading state information...
E: Unable to locate package google-cloud-sdk-gke-gcloud-auth-plugin
Error: Process completed with exit code 100.
```

I have a working workaround, i was just testing the different workarounds that people were posting to check which one i should go. 
my case is the google cloud auth plugin is not present anymore it looks like it.

**Comment by moffe42** [comment]
Looks like most of Github is having issues, https://www.githubstatus.com/, again...

**Comment by prateeksen** [comment]
#metoo

**Comment by MWLB** [comment]
#metoo

**Comment by drumst0ck** [comment]
#metoo

**Comment by AJulianah** [comment]
Same here

**Comment by xfr-nrb** [comment]
Workaround used (and working) using `ubuntu-20.04` instead of `ubuntu-latest` in `runs-on:`
Hope it helps ;-)

**Reply by drumst0ck** [reply]
Its works for me

**Reply by mucahitgurbuz** [reply]
This works!

**Reply by 0Dmitry** [reply]
tried setting it to ubuntu-23.10 but it seems to have stuck getting that image

**Reply by lilfrou** [reply]
we want the fix on ubuntu-latest ;)

**Reply by xfr-nrb** [reply]
@0Dmitry We cannot use any available ubuntu image versions cfr https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories

**Reply by Ahmed-M-Osman1** [reply]
works for me. Thanks

**Reply by nikharthakur-psi** [reply]
ubuntu-20.04 yes it works for me.

**Comment by bobrosoft** [comment]
Same 💢

**Comment by xinyue1128** [comment]
same

**Comment by Ahmed-M-Osman1** [comment]
same here.

**Comment by rob-safi** [comment]
same here

**Comment by acezen** [comment]
same

**Comment by pramanouski** [comment]
We are also seeing this

**Comment by user-cube** [comment]
Probably related to this incidents:
https://www.githubstatus.com/

![image](https://github.com/community/community/assets/33549232/def6766d-1fdc-44de-be9f-b80e274d3bfc)

**Comment by uyenzen2020** [comment]
same here

**Comment by nikharthakur-psi** [comment]
While attempting to install `npx playwright install-deps` in the latest Ubuntu environment within Azure Pipeline, I'm encountering the same issue.

**Reply by Ahmed-M-Osman1** [reply]
https://github.com/orgs/community/discussions/120966#discussioncomment-9212104

**Comment by jypuigbo** [comment]
It seems that part of the issue could be happening with microsoft or with ubuntu itself, and not github actions per se. 

I'm having this same issue, on a local dockerfile, when building a new image, and early this morning it was working perfectly fine. 

Does anyone know how to reach microsoft or ubuntu about it so that they realize there is an issue?

**Comment by trolleyman** [comment]
Getting this on my local machine right now when trying to apt update

**Comment by Polymere** [comment]
Same here, probably would not have happened if we all used 🪟 (which is the superior os for devs)

**Comment by VOVELEE** [comment]
Same here

**Comment by acollado2-cambridge** [comment]
Same here

**Comment by rbzinnia** [comment]
Facing this too. Hope MS fixes this soon.

**Comment by JonathanAsamoah** [comment]
Same here

**Comment by thivinanandh** [comment]
Facing same issues. However `20.04` seems to be working as of now.

**Comment by Teingi** [comment]
Same here

**Comment by shubham399** [comment]
same here, using 20.04 for now

**Comment by sripwoud** [comment]
In case some need urgently to fix their builds (especially in ci), remove the microsoft repository:
`sudo rm /etc/apt/sources.list.d/microsoft-prod.list`

e.g 
https://github.com/semaphore-protocol/semaphore/pull/751

**Comment by everfranca** [comment]
Up! 
Same here.

**Comment by gaelxcowi** [comment]
anyone a good workaround on managed blob-csi?

**Comment by muditjaju** [comment]
We  at [Mysa](https://mysa.io) are also facing the same issue!

**Comment by jypuigbo** [comment]
Nice, found a more befitting answer for my specific problem. 

https://stackoverflow.com/a/78377916

In summary, apparently it's a recurring issue with Microsoft, and: 

> All you can do is 1) report it to Microsoft e.g. via [GitHub issues on relevant repositories](https://github.com/microsoft/linux-package-repositories/issues), or 2) wait and try again later.

Advice: 
- For those for whom removing the update or `microsoft-prod.list` works, consider checking where or when you added microsoft tools for linux, clean it, remove it, and make sure this won't break your dependencies elsewere (It's a good time for a cleanup)
- For those who need to update in order to install some microsoft package, please, leave as much reports on relevant repos, to make sure it's detected as early as possible
- For those whom changing to ubuntu20.04 worked, I'd say we have 2 different issues here and maybe you want to investigate a bit more before switching OS version, right?

**Comment by thewh1teagle** [comment]
Same here, ubuntu 22.04

**Comment by DevendraTomar** [comment]
Same issue here, this is impacting our Databricks clusters (Runtime 13.3 LTS) while setting up MSSQL ODBC driver using init_scripts route.

**Comment by deavmi** [comment]
Yo, same fr I do be getting this

**Comment by leticiarodrigues5** [comment]
Same

**Comment by haseebq** [comment]
+1 Ubuntu 22.04

**Comment by jhonatan-colina** [comment]
+1 Ubuntu 22.04

**Comment by elinoyelbaum** [comment]
+1

**Comment by harlem88** [comment]
+1

**Comment by meisbokai** [comment]
+1

**Comment by jdanielnd** [comment]
+1

Using `ubuntu-latest` resolving to `22.04.4`

**Comment by Orphail** [comment]
+1

**Comment by JaLe29** [comment]
Playwright https://playwright.dev/ test failing due this issue :-(

```
name: Playwright tests

on:
  workflow_dispatch:
    inputs:
      username:
        description: 'Start test'
        default: false
        required: true
        type: boolean
  schedule:
    - cron: "0 1 * * *"
jobs:
  tests_e2e:
    name: Run end-to-end tests
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v3
    - uses: actions/setup-node@v3
    - name: Install dependencies
      run: yarn
    - name: Install playwright browsers
      run: npx playwright install --with-deps
    - name: Run tests
      run: yarn  test
```

**Reply by devandrews** [reply]
Yes, same for me

**Reply by shipplix** [reply]
try this: https://github.com/microsoft/playwright/issues/30503#issuecomment-2074783821

**Reply by Hirata-Matheus** [reply]
Yes, I got it too

**Comment by tkpitta** [comment]
Same error here, cmon Ubuntu devs (or microsoft package distribution)! We all need you!!

**Comment by AleksAlonso** [comment]
same here.

**Comment by ghzeni** [comment]
I'm less worried about how to fix this (specially since changing `runs-on: ubuntu-20.04` seems to do the trick) than about why did this happen at all?

**Reply by trolleyman** [reply]
It's probably because the repo is returning an empty file: https://github.com/orgs/community/discussions/120966#discussioncomment-9211978

No idea why that's the case, but yep that's why.

**Reply by ghzeni** [reply]
feels great to see sus stuff like this after the xz backdoor 💀

**Reply by jepio** [reply]
@ghzeni: see this https://github.com/microsoft/linux-package-repositories/issues/130#issuecomment-2074910628

**Comment by benner** [comment]
At least for me, action went successful.

**Comment by Lewenhaupt** [comment]
Seems to have been resolved

**Comment by jshossen** [comment]
Works for me: 
runs-on: ubuntu-20.04 instead of runs-on: latest in your .yml

**Comment by 24icewolf42** [comment]
Please don't judge me about security implications, but you can workaround until it is fixed by adding the following options to the `sources.list` entry: `allow-downgrade-to-insecure=yes allow-insecure=yes`
```bash
$ cat /etc/apt/sources.list.d/microsoft-prod.list
deb [arch=amd64 signed-by=/usr/share/keyrings/microsoft.gpg allow-downgrade-to-insecure=yes allow-insecure=yes] https://packages.microsoft.com/ubuntu/22.04/prod jammy main
```

BTW: It is already fixed, but keep in mind for the next time ;-)

**Comment by aditya-mangosciences** [comment, marked as answer]
This issue now resolved, We dont need to do anything

**Reply by jepio** [reply]
Root cause from the PMC team here: https://github.com/microsoft/linux-package-repositories/issues/130#issuecomment-2074910628

**Reply by jepio** [reply]
Also see this reply from @daviddavis ([link](https://github.com/orgs/community/discussions/120966#discussioncomment-9213550))
> Apologies for the disruption. It should be resolved now. Our signing service failed to sign the repo files and we'll rolling out a feature to check the repo metadata is signed properly before we publish it. If you all encounter any problems with packages.microsoft.com in the future, please reach out to https://github.com/microsoft/linux-package-repositories/issues.

**Comment by petervandenabeele** [comment]
My workflows work again without any changes (reran failed job successfully).

**Comment by daviddavis** [comment]
Apologies for the disruption. It should be resolved now. Our signing service failed to sign the repo files and we'll rolling out a feature to check the repo metadata is signed properly before we publish it. If you all encounter any problems with packages.microsoft.com in the future, please reach out to https://github.com/microsoft/linux-package-repositories/issues.

**Comment by jfuruness** [comment]
still getting this issue

**Reply by daviddavis** [reply]
We deployed some fixes this morning that ought to address the issue. Can you try again?

**Reply by jfuruness** [reply]
looks to be working now, thank you :)

**Comment by DevZank** [comment]
Me too =(

**Reply by jasonzio** [reply]
Are you still seeing this behavior? If this is a current issue (2024-04-29 or later) please report it at [our issues page here at github.com](https://github.com/microsoft/linux-package-repositories/issues)

**Reply by DavCyril** [reply]
Im currently facing this dilemma its been bugging me all day, and nothing i do seems to resolve it
